audio/out/internal.h

```diff
@@ -149,6 +148,9 @@ struct ao_driver {
     // Stop all audio playback, clear buffers, back to state after init().
     // Optional for pull AOs.
     void (*reset)(struct ao *ao);
+    // pull based: set pause state. Only called after start() and before reset().
+    //             The return value is ignored.
```

Comment:
thought: if this returns value shouldn't the core call `reset`?

Reply:
That will make things complicated, and require a new field to maintain whether the ao is paused using `set_pause` or `reset` like the `hw_paused` field in push-based ao. Giving `set_pause` is invoked outside `p->lock`, to achieve this we have to re-acquire it multiple times... I'd rather keep things simple. After all, `set_pause` can invoke `reset` itself if needed.

Reply:
Hmm, I guess that should be okay.